File: src/Domain/BoundedContext/AccountContext/Account.cs
using System;
using DrifterApps.Holefeeder.Domain.Enumerations;
using DrifterApps.Holefeeder.Domain.SeedWork;

namespace DrifterApps.Holefeeder.Domain.BoundedContext.AccountContext
{
    public class Account : Entity, IAggregateRoot
    {
        public Account(Guid id, AccountType type, string name, bool favorite, decimal openBalance, DateTime openDate,
            string description, bool inactive)
        {
            Id = id;
            Type = type;
            Name = name;
            Favorite = favorite;
            OpenBalance = openBalance;
            OpenDate = openDate;
            Description = description;
            Inactive = inactive;
        }

        public AccountType Type { get; }
        public string Name { get; }
        public bool Favorite { get; }
        public decimal OpenBalance { get; }
        public DateTime OpenDate { get; }
        public string Description { get; }
        public bool Inactive { get; }
    }
}

File: src/Holefeeder.Ui/lib/core/models/account.dart
import 'package:decimal/decimal.dart';
import 'package:holefeeder/core/enums/account_type_enum.dart';

class Account {
  final String id;
  final String name;
  final AccountType type;
  final Decimal openBalance;
  final DateTime openDate;
  final int transactionCount;
  final Decimal balance;
  final DateTime updated;
  final String description;
  final bool favorite;
  final bool inactive;

  const Account({
    required this.id,
    required this.name,
    required this.type,
    required this.openBalance,
    required this.openDate,
    required this.transactionCount,
    required this.balance,
    required this.updated,
    required this.description,
    required this.favorite,
    required this.inactive,
  });

  factory Account.fromJson(Map<String, dynamic> json) {
    return Account(
      id: json['id'] as String,
      name: json['name'] as String,
      type: AccountTypeExtension.fromString(json['type'] as String),
      openBalance: Decimal.parse(json['openBalance'].toString()),
      openDate: DateTime.parse(json['openDate'] as String),
      transactionCount: json['transactionCount'] as int,
      balance: Decimal.parse(json['balance'].toString()),
      updated: DateTime.parse(json['updated'] as String),
      description: json['description'] as String,
      favorite: json['favorite'] as bool,
      inactive: json['inactive'] as bool,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'type': type.toStringValue(),
      'openBalance': openBalance.toString(),
      'openDate': openDate.toIso8601String(),
      'transactionCount': transactionCount,
      'balance': balance.toString(),
      'updated': updated.toIso8601String(),
      'description': description,
      'favorite': favorite,
      'inactive': inactive,
    };
  }
}
